Integrations/CarbonBlackProtect/CarbonBlackProtect.py

```diff
@@ -0,0 +1,1105 @@
+    :param hr_title: Title of human readable
+    :param ec_key: Entry Context key
+    :return: EntryObject of the item
+    """
+    args = demisto.args()
+    id = args.get('id')
+    headers = args.get('headers')
+    raw_res = get_function(id)
+    ec = get_trasnformed_dict(raw_res, trans_dict)
+    hr = tableToMarkdown(hr_title, ec, headers, removeNull=True, headerTransform=pascalToSpace)
+    ec = {ec_key: ec} if ec else None
+    demisto.results(create_entry_object(raw_res, ec, hr))
+
+
+def remove_prefix(prefix, full_str):
+    """
+    Removes prefix from beginning of full_str if found
+    :param prefix: Prefix to remove from full_str
+    :param full_str: String to have its prefix removed
+    :return: full_str without the provided prefix
+    """
+    if full_str.startswith(prefix):
+        return full_str[len(prefix):]
+    return full_str
+
+
+def event_severity_to_dbot_score(severity):
+    """
+        Converts an severity int to DBot score representation
+        Event severity. Can be one of:
+        2 = Critical    -> 3
+        3 = Error       -> 0
+        4 = Warning     -> 2
+        5 = Notice      -> 2
+        6 = Info        -> 0
+        7 = Debug       -> 0
+
+        :type severity: ``int``
+        :param severity: Int representation of a severity
+
+        :return: DBot score representation of the severity
+        :rtype ``int``
+    """
+    if severity == '2':
+        return 3
+    elif severity in (4, 5):
+        return 2
+    return 0
+
+
+def cbp_date_to_timestamp(date):
+    """
+    Converts a date in carbon black's format to timestamp
+    :param date: Date string in cbp date format
+    :return: Timestamp of the given date
+    """
+    try:
+        ts = date_to_timestamp(date, date_format=CB_TIME_FORMAT)
+    except ValueError:
+        ts = date_to_timestamp(date, date_format=CB_NO_MS_TIME_FORMAT)
+    return ts
+
+
+def event_to_incident(event):
+    """
+        Creates an incident of a detection.
+
+        :type event: ``dict``
+        :param event: Single event object
+
+        :return: Incident representation of an event
+        :rtype ``dict``
+    """
+    incident = {
+        'name': 'CBP Event ID: ' + str(event.get('id')),
+        'occurred': str(event.get('timestamp')),
+        'rawJSON': json.dumps(event),
+        'severity': event_severity_to_dbot_score(event.get('severity'))
+    }
+    return incident
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Performs basic get request to get item samples
+    """
+    http_request('GET', '/computer?limit=-1')
+
+
+def search_file_catalog_command():
+    """
+    Searches for file catalog
+    :return: EntryObject of the file catalog
+    """
+    generic_search_command(
+        search_function=search_file_catalog,
+        trans_dict=FILE_CATALOG_TRANS_DICT,
+        hr_title='CarbonBlack Protect File Catalog Search',
+        ec_key='File(val.SHA1 === obj.SHA1)'
```

Comment:
will sha1 be present in any correct result?

Reply:
yes
![image](https://user-images.githubusercontent.com/20818773/56470501-9a928b80-644f-11e9-8f9f-3a7b224aa82e.png)